src/main/java/com/dongxuexidu/douban4j/provider/OAuthDoubanProvider.java

```diff
@@ -148,7 +148,7 @@ public AccessToken tradeAccessTokenWithCode (String code) throws DoubanException
       Map<String,String> params = new HashMap<String, String>();
       params.put("client_id", this.apiKey);
       params.put("client_secret", this.secretKey);
-      params.put("redirect_uri", DefaultConfigs.ACCESS_TOKEN_REDIRECT_URL);
+      params.put("redirect_uri",  this.redirectUrl);
```

Comment:
I think you've done this check in OAuthDoubanProvider.getGetCodeRedirectUrl() method like  :
public String getGetCodeRedirectUrl() {
		if (this.redirectUrl == null || this.redirectUrl.isEmpty()) {
			logger.warning("Redirect url cannot be null or empty, did you forget to set it?");
			return null;
		}